### PR TITLE
IGAPP-1097: Sort Pois by Distance

### PR DESCRIPTION
**Pull request by sarahsporck**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Review by Cmd190 — APPROVED** [review]
Tested on firefox and android. Looks like it works.

File: api-client/src/utils/geoJson.ts [hunk 1]
```diff
@@ -8,17 +8,26 @@ export const embedInCollection = (features: PoiFeature[]): PoiFeatureCollection
   features,
 })
 
-export const prepareFeatureLocation = (poi: PoiModel, userLocation: LocationType | null): PoiFeature | null => {
+export const prepareFeatureLocation = (poi: PoiModel, userLocation: LocationType | null): PoiFeature => {
   const { featureLocation } = poi
-  if (userLocation && featureLocation) {
-    const distanceValue = distance(userLocation, featureLocation.geometry.coordinates).toFixed(1)
+  if (userLocation) {
+    const distanceValue = distance(userLocation, featureLocation.geometry.coordinates).toFixed(0)
```
Comment (steffenkleinle) [inline api-client/src/utils/geoJson.ts:14]: Why did you choose to change this? I think we should perhaps differ between small values < 1 or 2 (one digit after comma) and bigger ones (no digit after comma).

File: api-client/src/utils/geoJson.ts [hunk 2]
```diff
@@ -8,17 +8,26 @@ export const embedInCollection = (features: PoiFeature[]): PoiFeatureCollection
...
+  if (userLocation) {
+    // sort by distance to current location
+    return poiFeatures.sort(
+      (poi1: PoiFeature, poi2: PoiFeature) =>
+        parseFloat(poi1.properties.distance ?? '0') - parseFloat(poi2.properties.distance ?? '0')
+    )
+  }
+  return poiFeatures.sort((poi1: PoiFeature, poi2: PoiFeature) =>
+    poi1.properties.title.localeCompare(poi2.properties.title)
+  )
```
Comment (steffenkleinle) [inline api-client/src/utils/geoJson.ts:32]: No need for the type annotations:
```suggestion
   if (userLocation) {
    // sort by distance to current location
    return poiFeatures.sort(
      (poi1, poi2) => parseFloat(poi1.properties.distance ?? '0') - parseFloat(poi2.properties.distance ?? '0')
    )
  }
  return poiFeatures.sort((poi1, poi2) => poi1.properties.title.localeCompare(poi2.properties.title))
```

**Review by steffenkleinle — APPROVED** [review]
Tested on firefox

File: api-client/src/utils/geoJson.ts [hunk 3]
```diff
@@ -8,17 +8,26 @@ export const embedInCollection = (features: PoiFeature[]): PoiFeatureCollection
...
-export const prepareFeatureLocations = (pois: Array<PoiModel>, userLocation: LocationType | null): PoiFeature[] =>
-  pois
-    .map(poi => prepareFeatureLocation(poi, userLocation))
-    .filter((feature): feature is PoiFeature => !!feature)
-    .sort((poi1: PoiFeature, poi2: PoiFeature) => poi1.properties.title.localeCompare(poi2.properties.title))
+export const prepareFeatureLocations = (pois: Array<PoiModel>, userLocation: LocationType | null): PoiFeature[] => {
+  const poiFeatures = pois.map(poi => prepareFeatureLocation(poi, userLocation))
+
+  if (userLocation) {
+    // sort by distance to current location
```
Comment (SteffiStuffel) [inline api-client/src/utils/geoJson.ts:24]: I think this comment is not needed.

**Review by SteffiStuffel — APPROVED** [review]
Looks good, not tested.

File: api-client/src/utils/geoJson.ts (hunk 1 above)
Comment (sarahsporck) [reply api-client/src/utils/geoJson.ts:14]: oh. wasn't intentional. will revert.

**Pull request merged**